src/node_nanomsg.cc

```diff
@@ -125,6 +130,41 @@ NAN_METHOD(Getsockopt) {
     NanReturnValue(obj);
 }
 
+NAN_METHOD(Setopt) {
+  NanScope();
+
+  int level = args[1].integer;
+  int option = args[2].integer;
+
+  if(option == NN_SOCKET_NAME){
+    utf8 str(args[3]);
+    ret(NanNew<Number>(nn_setsockopt(S, level, option, *str, str.length())));
+  } else {
+    int optval = args[3].integer;
+    ret(NanNew<Number>(nn_setsockopt(S, level, option, &optval, sizeof(optval))));
+  }
+}
+
+NAN_METHOD(Getopt) {
+  NanScope();
+
+  int optval[64];
+  int option = args[2].integer;
+  size_t optsize = sizeof(optval);
+
+  //check if the function succeeds
+  if(nn_getsockopt(S, args[1].integer, option, optval, &optsize) == 0){
```

Comment:
[We were recently mucking with the exported symbols](https://github.com/nickdesaulniers/node-nanomsg/pull/43), so it's possible we missed one.

This function has bigger issues, do to the fact that set/getsocketopts can work on a single int, or a `char[]` when the level is `NN_SUB` and the option is `NN_SUB_SUBSCRIBE`.  Please see the examples sections from [set](http://nanomsg.org/v0.2/nn_setsockopt.3.html) and [get](http://nanomsg.org/v0.2/nn_getsockopt.3.html).  See also [usage](https://github.com/nanomsg/nanomsg/search?utf8=%E2%9C%93&q=nn_getsockopt) within the lib.

@sustrik, under what conditions does `nn_getsocketopts` expect to be passed a `char[]`?